akka-http-core/src/main/scala/akka/http/model/MediaType.scala

```diff
@@ -200,28 +200,47 @@ class MultipartMediaType private[http] (_value: String, _subType: String, _param
 }
 
 sealed abstract class NonMultipartMediaType private[http] (_value: String, _mainType: String, _subType: String,
-                                                           _compressible: Boolean, _binary: Boolean,
+                                                           _compressible: Boolean, _encoding: MediaType.Encoding,
                                                            _fileExtensions: immutable.Seq[String],
                                                            _params: Map[String, String])
-  extends MediaType(_value)(_mainType, _subType, _compressible, _binary, _fileExtensions, _params) {
-  private[http] def this(mainType: String, subType: String, compressible: Boolean, binary: Boolean, fileExtensions: immutable.Seq[String]) =
-    this(mainType + '/' + subType, mainType, subType, compressible, binary, fileExtensions, Map.empty)
+  extends MediaType(_value)(_mainType, _subType, _compressible, _encoding, _fileExtensions, _params) {
+  private[http] def this(mainType: String, subType: String, compressible: Boolean, encoding: MediaType.Encoding,
+                         fileExtensions: immutable.Seq[String]) =
+    this(mainType + '/' + subType, mainType, subType, compressible, encoding, fileExtensions, Map.empty)
   def withParams(params: Map[String, String]) =
-    MediaType.custom(mainType, subType, compressible, binary, fileExtensions, params)
+    MediaType.custom(mainType, subType, compressible, encoding, fileExtensions, params)
 }
 
 object MediaType {
+  sealed abstract class Encoding(val charset: Option[HttpCharset])
+  object Encoding {
+    /**
+     * Indicates that the media type is non-textual and a character encoding therefore has no meaning.
+     */
+    case object Binary extends Encoding(None)
+
+    /**
+     * Indicates that the media-type allow for flexible character encoding through a `charset` parameter.
+     */
+    case object Open extends Encoding(None)
+
+    /**
+     * Indicates that a media-type is textual and mandates a clearly defined character encoding.
+     */
+    final case class Fixed(cs: HttpCharset) extends Encoding(Some(cs))
+  }
+
   /**
    * Create a custom media type.
    */
-  def custom(mainType: String, subType: String, compressible: Boolean = false, binary: Boolean = false,
-             fileExtensions: immutable.Seq[String] = Nil, params: Map[String, String] = Map.empty,
-             allowArbitrarySubtypes: Boolean = false): MediaType = {
+  def custom(mainType: String, subType: String, compressible: Boolean = false,
+             encoding: MediaType.Encoding = Encoding.Binary, fileExtensions: immutable.Seq[String] = Nil,
```

Comment:
@sirthias thanks for the explanation. I guess it's only a default value so people can still change it easily for their own custom media-types. My slight preference would probably be to remove the default value but it's a tough call.